### PR TITLE
Improve User Guide and Developer Guide: Refine use cases and manual testing

### LINKED ISSUE #189: [Enhancement] Improve User Guide and Developer Guide: Refine use cases and manual testing
Improves the quality and clarity of the User Guide (UG) and Developer Guide (DG), focusing on use cases and manual testing documentation.

### PR DESCRIPTION
**Pull request by savagexr3**
## Overview
This PR improves the quality and clarity of the User Guide (UG) and Developer Guide (DG), focusing on use cases and manual testing documentation.

### Use Cases
- Refactored all use cases to follow a consistent format (MSS + Extensions)
- Fixed grammar, wording, and step clarity
- Ensured proper abstraction (removed implementation-level details)
- Split overly broad use cases (e.g. tag operations) into separate, focused ones
- Added missing use cases:
  - Restore contact
  - View deleted contacts
  - Find contacts by name
  - Show contacts by tag
- Added relevant edge cases (invalid input, duplicates, empty results)

### Manual Testing
- Improved structure and readability of test cases
- Standardised formatting (`Test case`, `Expected`)
- Added test coverage for core commands (add, edit, delete)
- Included edge cases (invalid inputs, boundary values, corrupted data file)

### Consistency Improvements
- Standardised naming across documentation (`Clinkedin`)
- Improved formatting and readability throughout UG and DG

## Why
- Enhances documentation clarity and usability for users and testers
- Improves alignment with CS2103 documentation standards
- Ensures comprehensive coverage of core features and edge cases

## Testing
- Documentation changes reviewed manually
- Verified consistency across UG and DG

## Issues
- Closes #189

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (rxlee04) [inline docs/DeveloperGuide.md]: need update because now have unique tag list

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -677,124 +817,131 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-- Transformed tags into fully managed entities
-- Implemented deleted-person tracking and restoration
-- Improved validation and usability (e.g., clickable links and standardized color input)  
+### Additional Testing
 
-## **Appendix: Planned Enhancements**
-These enhancements are planned for future iterations:
-- Improve tag restoration behaviour: when restoring a deleted contact, tags that were renamed will be correctly mapped to their updated names instead of being removed.
+Testers should also explore:
```
Comment (rxlee04) [inline docs/DeveloperGuide.md]: planned enhancement section shouldn't be removed either

**Comment by savagexr3** [comment]
This PR is being closed as the current master branch has diverged significantly, resulting in extensive overwrites and conflicts. A fresh PR based on the latest master would be more appropriate.
